src/utilities/path.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { describe, expect, test } from 'vitest'
+import { optional } from '@/types'
+import { path } from '@/utilities'
+
+describe('path', () => {
+  test('given path without params, returns empty object', () => {
+    const response = path('/string/example/without/params', {})
+
+    expect(response.params).toMatchObject({})
+  })
+
+  test('given path with simple params, returns each param name as type String', () => {
+    const response = path('/parent/:parentId/child/:childId', {})
+
+    expect(response.params).toMatchObject({
+      parentId: [String],
+      childId: [String],
+    })
+  })
+
+  test('given path with optional params, returns each param name as type String with optional', () => {
+    const response = path('/parent/:?parentId/child/:?childId', {})
+
+    expect(JSON.stringify(response.params)).toMatch(JSON.stringify({
```

Comment:
curious why you had to stringify here but not in the other tests? Just because of the optional function?

Reply:
if I don't stringify, the `optional(String)` just ends up as `[Function Boolean]`. This is the only way I could find to verify that the param will be "Optional" as we've defined it with the business logic found in optional function